### PR TITLE
CAD 484 | trace forwarding:  named pipes for windows

### PR DESCRIPTION
**Pull request by deepfire**
1. Copy over the `Ouroboros.Network.Snocket` module from `ouroboros-network-framework`
2. Reimplement the forwarder and acceptor in terms of snockets, thereby automagically gaining named pipe support on Windows.
3. Syncs all dependency versions to that of `cardano-node`.
4. Fix evaluation errors in Nix code.

checklist
---------

- [x] compiles (`cabal v2-build` or `stack build`)
- [x] tests run successfully (`cabal v2-test` or `stack test`)
- [ ] documentation added
- [ ] link to an issue
- [ ] add milestone (the current sprint)

### CODE REVIEW

File: iohk-monitoring/src/Cardano/BM/Snocket.hs [hunk 1]
```diff
@@ -0,0 +1,429 @@
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Snocket.hs:5]: reference where this is coming from ...

File: stack.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
-resolver: lts-13.26
+resolver: https://raw.githubusercontent.com/input-output-hk/cardano-prelude/3ac22a2fda11ca7131a011a9ea48fcbfdc26d6b3/snapshot.yaml
```
Comment (CodiePP) [inline stack.yaml:1]: we should not depend on cardano-prelude

File: iohk-monitoring/src/Cardano/BM/Snocket.hs (hunk 1 above)
Comment (deepfire) [reply iohk-monitoring/src/Cardano/BM/Snocket.hs:8]: Done.

File: stack.yaml (hunk 2 above)
Comment (deepfire) [reply stack.yaml]: Done, let's hope it builds with `lts-13.26`.

**Comment by deepfire** [comment]
@CodiePP, do we have any action items here?

**Comment by CodiePP** [comment]
compiled on Windows using `stack`

**Comment by deepfire** [comment]
:tada:

**Pull request merged**